kratos/includes/constitutive_law.h

```diff
@@ -867,7 +867,7 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
      * @param rShapeFunctionsValues the shape functions values in the current integration point
      * @param the current ProcessInfo instance
      */
-    KRATOS_DEPRECATED_MESSAGE("Please do not use this method\"")
+    KRATOS_DEPRECATED_MESSAGE("Please do not use this method - There is no equivalent for this\"")
```

Comment:
if there is no equivalent to this the I think that some refactoring is required

@loumalouomega @AlejandroCornejo how do you update the plasticity-related values in your plasticity-laws?

Reply:
You are supposed to save for the converged problem, right @AlejandroCornejo ?

Reply:
I need this in my algorithm to check if my element was in the plastic regime in the last non-lin iteration step. I cannot say exactly why this was necessary as I wrote that some months ago. But I am not updating stresses or strains in this function

Reply:
why do you want to get rid of the functionality that the constitutive law can call `FinalizeNonLinearIteration` ? Does it crash something else?

Reply:
No, it is only old functions maintained for retrocompatibility. @RiccardoRossi we can replace them with InitializeMaterialReponseInNonLinearIteration and FinalizeMaterialReponseInNonLinearIteration. In any case I am convinced that you can find a way to maintain the behaviour with the current non deprectaed API

Reply:
I think it should be possible to do it without the functions since it also works for the other plasticity-laws, but we need some time to look into it in detail